cpp/ql/test/library-tests/deduction_guides/test.cpp

```diff
@@ -0,0 +1,18 @@
+// semmle-extractor-options: -std=c++20
+
+template<typename T>
+struct C {
+    C(const T);
+};
+
+C(const double)  -> C<int>;
```

Comment:
Why aren't we finding this? It's the one that should be selected by `new C(0.0);`, so it's not even unused, which would have been my best guess.

Are we missing the deduction guide itself, or its template class? I.e. can we add a separate query that just does `from DeductionGuide d select d`?

Reply:
We are finding it, but the frontend doesn't assign a proper location to it. This is the following line in the expected test results:
```
| file://:0:0:0:0 | C | test.cpp:4:8:4:8 | C<T> |
```

Reply:
I was assuminig that's the [copy deduction candidate](https://en.cppreference.com/w/cpp/language/class_template_argument_deduction#Implicitly-generated_deduction_guides) (which indeed wouldn't have a location)?
> In any case, an additional fictional function template derived as above from a hypothetical constructor C(C) is added, called the copy deduction candidate.

Reply:
(I guess an easy way to check would be to remove `C(const double)  -> C<int>;` and see if `| file://:0:0:0:0 | C | test.cpp:4:8:4:8 | C<T> |` is still there.

Reply:
The copy deduction one will be the unnamed one.

Reply:
🤔 But the unnamed one does have a location, and it's on line 5, where the `C(const T);` constructor is?

Basically, we have 4 deduction guides here and only 3 results lines (copy deduction guide, the guide for the `C(const T);` constructor, and the two explicit guides).

Reply:
Then the copy one is likely missing. Can we side-step the discussion about missing ones for now, and open an internal issue for this? This is pretty much out-of-scope for this PR.

Reply:
👍 I agree based on the new test results. Could you please report:
 1. The missing locations for non-template deduction guides
 2. The missing copy deduction guide?